internal/credentials/credentials.go

```diff
@@ -100,6 +101,15 @@ func (k *kubernetesDatabase) Get(
 ) (Credentials, bool, error) {
 	creds := Credentials{}
 
+	// If we are dealing with an insecure HTTP endpoint (of type Git),
```

Comment:
See [comment](https://github.com/akuity/kargo/issues/872#issuecomment-1957427712)

Ok with the breaking change here. We can get away with it for a short while longer and should take advantage while we can.

Reply:
Apart from this, the rest LGTM